### ISSUE TITLE
Training script required to retrain the network on different data sets

### ISSUE THREAD

**Issue by ashishtibrewal** [luizgh/sigver_wiwd#7]
I was going through the source files and could not find the training script. I would like to retrain the network on a few different data sets as the accuracy for the pre-trained model(s) on the data I have is not very good, and I think there can be significant improvements after re-training using the data I am working with. Could you please provide the training script that was used for the published work (and used to train the models provided) or a brief guide stating how to go about doing this would be really helpful? Thank you for your assistance.

**Comment by luizgh** [comment]
Hi,

We are planing on releasing all the code in the future, after we finish some papers we are working on (I don't have a date for that). For re-training the model, you first need to process your dataset (using the provided functions) so that all signatures have the same size. You should have a tensor `x` with all signatures (M x 1 x H x W), where M is the number of images, and H and W are their size. You should also have a vector `y` (size M) that contains the labels (which user created the signature). You can then take the signet model, and add a new classification layer. Suppose you have 100 writers, you do as follows:

```python
net['fc3'] = DenseLayer(net['fc2'], num_units=100)
```

You can then define a loss function and train the network using stochastic gradient descent. There are many tutorials available online on how to do so, here is one I created on how to use lasagne: http://luizgh.github.io/libraries/2015/12/08/getting-started-with-lasagne/

**Comment by ashishtibrewal** [comment]
Thanks for the quick reply. This is very helpful. :)

**Comment by ashishtibrewal** [comment]
What is the (default) value of lambda in the signet.pkl model? In the three models provided, two of those have a lambda value specified in the file name but one of them doesn't. I am guessing it is 0.99 (this guess is based on the values stated in your paper).

Also, on Table 4 (Page 18), should the value for the testing set for the CEDAR dataset be 12 genuine signatures rather than 10 as the training set consists of 12 samples, and each user has a total of 24 genuine signatures as stated in Table 3?

**Comment by luizgh** [comment]
There is a big distinction between SigNet and SigNet-F. SigNet used only genuine signatures for training (section 3.1 on the paper), while SigNet-F uses genuine signatures and forgeries (from the development set, see section 3.2.2). The lambda hyperparameter is only used in the SigNet-F formulation (to trade-off between the two objectives), and therefore there is no value of lambda for signet.pkl.

Regarding table 4, you are correct. The table should state that 12 genuine samples were used for testing the system
